Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -505,13 +469,15 @@ void CGPathCloseSubpath(CGMutablePathRef path) {
     RETURN_IF(!path);
 
     // Move the current point to the starting point since the line is closed.
-    path->SetCurrentPoint(path->GetStartingPoint());
-    path->EndFigure(D2D1_FIGURE_END_CLOSED);
+    // path->SetCurrentPoint(path->GetStartingPoint());
+    if (!CGPointEqualToPoint(path->GetStartingPoint(), path->GetCurrentPoint())) {
+        CGPathAddLineToPoint(path, nullptr, path->GetStartingPoint().x, path->GetStartingPoint().y);
+    }
+    path->EndFigure(D2D1_FIGURE_END_OPEN); // CLOSED);
 }
```

Comment:
Ignore me, i am braindead.

---
In reply to: [88095262](https://github.com/Microsoft/WinObjC/pull/1384#discussion_r88095262) [](ancestors = 88095262)